File: algorithms/Implement Rand() Using Rand()/Implement Rand() Using Rand().cpp
class Solution {
public:
    int rand10() {
        while(true)
        {
            int col = rand7();
            int last_row_tak = (rand7() - 1)*7;
            int val = last_row_tak + col;
            if(val<=40)
            {
                return (val-1)%10 + 1;
            }
        }
    }
};

File: scripts/algorithms/Implement Rand() Using Rand()/Implement Rand() Using Rand().py
class Solution:
    def rand10(self):
        """
        :rtype: int
        """
        x = rand7()
        y = rand7()
        pos = (x - 1) * 7 + y
        if pos > 40:
            return self.rand10()
        return (pos % 10) + 1
		```